feat(traefik)!: use cloudflare tunnel

dargmuesli opened the pull request:
We could shut off port 80 and 443 fully when making use of cloudflare's tunnels.

~~The initial commit sets the host ip to localhost, but~~ As `host_ip` is seemingly not available for docker swarm, which would make sense in a distributed environment, most likely `cloudflared` should be added as service to the stack, allowing us to completely drop the port definition for traefik on production.

dargmuesli committed:
feat(traefik)!: use localhost

semantic-release-bot committed:
chore(release): 6.0.0-beta.1 [skip ci]

## [6.0.0-beta.1](https://github.com/maevsi/maevsi_stack/compare/5.2.3...6.0.0-beta.1) (2024-05-23)

### ⚠ BREAKING CHANGES

* **traefik:** use localhost

### Features

* **traefik:** use localhost ([34f0902](https://github.com/maevsi/maevsi_stack/commit/34f09025ab29e971414c88712fac62ab61d1c53a))

### Bug Fixes

* **deps:** update ghcr.io/maevsi/maevsi to v4.9.0 ([5d12b12](https://github.com/maevsi/maevsi_stack/commit/5d12b12f23fbee322b006b6d89100f9add33cab5))

dargmuesli committed:
fix: revert "feat(traefik)!: use localhost"

This reverts commit 34f09025ab29e971414c88712fac62ab61d1c53a.

dargmuesli committed:
Merge branch 'master' into beta

dargmuesli committed:
fix(ci): update dargmuesli/github-actions to v2.1.1-beta.1

dargmuesli committed:
ci: specify pull request types instead of late checking

semantic-release-bot committed:
chore(release): 6.0.0-beta.2 [skip ci]

## [6.0.0-beta.2](https://github.com/maevsi/maevsi_stack/compare/6.0.0-beta.1...6.0.0-beta.2) (2024-05-23)

### Bug Fixes

* **ci:** update dargmuesli/github-actions to v2.1.1-beta.1 ([60cb03d](https://github.com/maevsi/maevsi_stack/commit/60cb03deba0dc7ac9123435623a2c176a5f4a6c5))
* **jobber:** deactivate virtual environment when done ([79e1de3](https://github.com/maevsi/maevsi_stack/commit/79e1de379c6eaab345af2e0c57c65213d08d1e88))
* revert "feat(traefik)!: use localhost" ([71a2c01](https://github.com/maevsi/maevsi_stack/commit/71a2c012f28420c3f9cea63d3a95b4654fa74210))

dargmuesli committed:
fix(ci): test pull request

dargmuesli committed:
fix(ci): test pull request

dargmuesli committed:
fix: revert "ci: specify pull request types instead of late checking"

This reverts commit c2bb90488be774e923fa395f83b4a334935cd7fe.

dargmuesli committed:
ci: skip build

dargmuesli committed:
chore(deps): update dargmuesli/github-actions to v2.2.0-beta.3

dargmuesli committed:
chore(deps): update dargmuesli/github-actions to v2.2.0-beta.4

dargmuesli committed:
fix(ci): test pull request

dargmuesli committed:
Merge pull request #131 from dargmuesli/test

fix(ci): test pull request

dargmuesli committed:
chore(deps): update dargmuesli/github-actions to v2.2.0-beta.5

semantic-release-bot committed:
chore(release): 6.0.0-beta.3 [skip ci]

## [6.0.0-beta.3](https://github.com/maevsi/maevsi_stack/compare/6.0.0-beta.2...6.0.0-beta.3) (2024-05-27)

### Bug Fixes

* **ci:** test pull request ([102c2b0](https://github.com/maevsi/maevsi_stack/commit/102c2b0b6aa68b9ea6dee42de387592776347312))
* **ci:** test pull request ([d001388](https://github.com/maevsi/maevsi_stack/commit/d0013886ae23a9d1106ac7eae09f8ba2f2b6cff9))
* **ci:** test pull request ([8dea724](https://github.com/maevsi/maevsi_stack/commit/8dea724f65180dcec9e0a03895cc729d178430a0))
* revert "ci: specify pull request types instead of late checking" ([7b47d0e](https://github.com/maevsi/maevsi_stack/commit/7b47d0ec8eee416a3c30d75703ae989ebbdc108b))

mrlnstk committed:
feat(infra): remove redirect and use port 80 for incoming connections

semantic-release-bot committed:
chore(release): 6.0.0-beta.4 [skip ci]

## [6.0.0-beta.4](https://github.com/maevsi/maevsi_stack/compare/6.0.0-beta.3...6.0.0-beta.4) (2024-06-04)

### Features

* **infra:** remove redirect and use port 80 for incoming connections ([353dade](https://github.com/maevsi/maevsi_stack/commit/353dade03ccb22ac5ee281e60c852ee3213d79d6))

mrlnstk committed:
fix(traefik): remove host_ip property

semantic-release-bot committed:
chore(release): 6.0.0-beta.5 [skip ci]

## [6.0.0-beta.5](https://github.com/maevsi/maevsi_stack/compare/6.0.0-beta.4...6.0.0-beta.5) (2024-06-04)

### Bug Fixes

* **traefik:** remove host_ip property ([dd6fcaa](https://github.com/maevsi/maevsi_stack/commit/dd6fcaac10a51524af77ccb8a64a0bdc22d2921b))

mrlnstk committed:
feat(prod): add cloudflared service to production deployment

semantic-release-bot committed:
chore(release): 6.0.0-beta.6 [skip ci]

## [6.0.0-beta.6](https://github.com/maevsi/maevsi_stack/compare/6.0.0-beta.5...6.0.0-beta.6) (2024-06-06)

### Features

* **prod:** add cloudflared service to production deployment ([2c04195](https://github.com/maevsi/maevsi_stack/commit/2c041955e8dd05bcf0c7faa59ba70988b4a29dd0))

mrlnstk committed:
feat(production): add command to cloudflare service

semantic-release-bot committed:
chore(release): 6.0.0-beta.7 [skip ci]

## [6.0.0-beta.7](https://github.com/maevsi/maevsi_stack/compare/6.0.0-beta.6...6.0.0-beta.7) (2024-06-06)

### Features

* **production:** add command to cloudflare service ([f822483](https://github.com/maevsi/maevsi_stack/commit/f822483cc6ee7217dbd3675fffd6552c050dfb80))

mrlnstk committed:
feat(cloudflared): add environment variable

semantic-release-bot committed:
chore(release): 6.0.0-beta.8 [skip ci]

## [6.0.0-beta.8](https://github.com/maevsi/maevsi_stack/compare/6.0.0-beta.7...6.0.0-beta.8) (2024-06-06)

### Features

* **cloudflared:** add environment variable ([868ba1c](https://github.com/maevsi/maevsi_stack/commit/868ba1c7476ab0f9152a576ef737b662892b9b3b))

mrlnstk committed:
fix(traefik): remove whole port definition on production

semantic-release-bot committed:
chore(release): 6.0.0-beta.9 [skip ci]

## [6.0.0-beta.9](https://github.com/maevsi/maevsi_stack/compare/6.0.0-beta.8...6.0.0-beta.9) (2024-06-06)

### Bug Fixes

* **traefik:** remove whole port definition on production ([31710ee](https://github.com/maevsi/maevsi_stack/commit/31710eeb71603ab2ceea186c9ebdf31d223f3cff))

dargmuesli commented on src/development/stack.yml:
```diff
@@ -99,7 +99,6 @@ services:
     deploy:
       labels:
         - traefik.enable=true
-        - traefik.http.routers.adminer.middlewares=redirectscheme
```
I think those should have a `#DARGSTACK-REMOVE` instead of being removed, no?

myyxl replied:
idk, is https also used in dev mode? if not it's not needed at all

dargmuesli replied:
yup

dargmuesli commented on src/development/stack.yml:
```diff
@@ -428,15 +415,11 @@ services:
-        protocol: tcp
-        published: 80
-        target: 80
-      - mode: host
-        protocol: tcp
-        published: 443
-        target: 443
```
I think we need to keep `443` for development.

dargmuesli commented on src/production/production.env.template:
```diff
@@ -1,4 +1,5 @@
-TRAEFIK_ACME_PROVIDER=
\ No newline at end of file
+TRAEFIK_ACME_PROVIDER=
+TUNNEL_TOKEN=
```
```suggestion
CLOUDFLARED_TUNNEL_TOKEN=
```

dargmuesli replied:
:heart:

dargmuesli commented on src/production/production.yml:
```diff
@@ -23,6 +23,12 @@ services:
+    environment:
+      TUNNEL_TOKEN: ${TUNNEL_TOKEN}
+    image: cloudflare/cloudflared
+    restart: unless-stopped
```
```suggestion
```
This line is not needed for docker stack, this is the default behavior which can't be changed afaik.

mrlnstk committed:
refactor(cloudflared): rename env variable and remove restart option

mrlnstk committed:
refactor(https): use dargstack remove to keep https in development stack

mrlnstk committed:
fix(redirect): re-add redirect regex

semantic-release-bot committed:
chore(release): 6.0.0-beta.10 [skip ci]

## [6.0.0-beta.10](https://github.com/maevsi/maevsi_stack/compare/6.0.0-beta.9...6.0.0-beta.10) (2024-06-06)

### Bug Fixes

* **redirect:** re-add redirect regex ([b840c59](https://github.com/maevsi/maevsi_stack/commit/b840c591928fd2aa40128d2ac1ddef3f1a9e7751))

mrlnstk reviewed (commented):
LGTM

dargmuesli committed:
fix(traefik): do not remove entrypoint in production

semantic-release-bot committed:
chore(release): 6.0.0-beta.11 [skip ci]

## [6.0.0-beta.11](https://github.com/maevsi/maevsi_stack/compare/6.0.0-beta.10...6.0.0-beta.11) (2024-06-06)

### Bug Fixes

* **traefik:** do not remove entrypoint in production ([ce0df8e](https://github.com/maevsi/maevsi_stack/commit/ce0df8eb2fec054b98748c607eabe77fadbd7cc0))

dargmuesli committed:
Merge branch 'master' into beta

semantic-release-bot committed:
chore(release): 6.0.0-beta.12 [skip ci]

## [6.0.0-beta.12](https://github.com/maevsi/maevsi_stack/compare/6.0.0-beta.11...6.0.0-beta.12) (2024-06-06)

### Bug Fixes

* **deps:** update ghcr.io/maevsi/maevsi to v4.9.3 ([2137c30](https://github.com/maevsi/maevsi_stack/commit/2137c308d6233a2d122b82d606ebabee7498e6ba))
